cmd/gen-go-sample/main.go

```diff
@@ -332,7 +333,15 @@ func (g *generator) genSample(ifaceName string, methConf GAPICMethod, regTag str
 	if len(argNames) > 0 {
 		var sb strings.Builder
 		for i, n := range argNames {
-			fmt.Fprintf(&sb, ", %s %s", n, pbinfo.GoTypeForPrim[argTrees[i].typ.prim])
+			if e, ok := argTrees[i].typ.desc.(*descriptor.EnumDescriptorProto); ok {
+				en, err := enumType(g.descInfo, e, g)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintf(&sb, ", %s %s", n, en)
+			} else {
+				fmt.Fprintf(&sb, ", %s %s", n, pbinfo.GoTypeForPrim[argTrees[i].typ.prim])
```

Comment:
Also, separate comment, I'm not super familiar with the `argTrees` type and the guarantees around `.typ.prim` but what if there is typo and the value doesn't exist? Will it print the `nil`? Or should we be handling that here (or maybe in arg parsing?)

Reply:
Added an error handling statement at the end. Nice catch!